### PR TITLE
[onert] Allow using I/O buffer for all models

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit allows using I/O buffers for all models on multimodel package. 
Intermediate executor output buffer should be used for next model executor.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

### CODE REVIEW

**Comment by hseok-oh** [comment]
@chunseoklee Could you check this PR works on example model?

**Comment by chunseoklee** [comment]
> @chunseoklee Could you check this PR works on example model?

I have to make a new nnpkg w/o need of type conversion(uint8->f32). Current target model consists of 1) trix model and 2) f32 circle model. I will update the result after creating new nnpkg. 

Anyway, it works w/o error with current model though it is weird that the size mismatch between trix's output and  circle's input does not occur any segfault.

**Comment by hseok-oh** [comment]
I confirmed that previous test packages (https://github.com/Samsung/ONE/issues/9646#issuecomment-1232445122, https://github.com/Samsung/ONE/issues/9655#issuecomment-1280216480) are passed.

**Comment by chunseoklee** [comment]
>  I will update the result after creating new nnpkg.

Execution passed. Need time to create verification value

File: runtime/onert/core/src/exec/Executors.cc [hunk 1]
```diff
@@ -97,99 +97,64 @@ void Executors::checkSupportedMultimodel() const
...
-      if (std::find_if(edges.begin(), edges.end(), check_from_edge) == edges.end())
-      {
-        throw std::runtime_error{"NYI: Unsupported model edge pattern"};
-      }
-    }
-  }
-
   // Assumption: edges
-  //   x:0:z -> x+1:0:z  (0 < x < n-2)
+  //   x:0:z -> x+1:0:z'
```
Comment (glistening) [inline runtime/onert/core/src/exec/Executors.cc:101]:
```suggestion
  //   x:0:z -> x+1:0:z
```

**Review by glistening — APPROVED** [review]
LGTM

File: runtime/onert/core/src/exec/Executors.cc (hunk 1 above)
Comment (hseok-oh) [reply runtime/onert/core/src/exec/Executors.cc:101]: `from.IOIndex` and `to.IOIndex` can be different on this commit.

**Review by jyoungyun — APPROVED** [review]

**Comment by chunseoklee** [comment]
> Need time to create verification value

Value matched with nnpkg of two F32 circle models.

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**